src/main/java/com/spotify/github/v3/clients/RepositoryClient.java

```diff
@@ -512,4 +627,70 @@ private String getContentPath(final String path, final String query) {
     }
     return String.format(CONTENTS_URI_TEMPLATE, owner, repo, path, query);
   }
+
+  private CompletableFuture<Commit> handleGetCommit(final Reference referenceResponse) {
+    ReferenceObject referenceObject = referenceResponse.object();
+    if (referenceObject == null) throw new GithubException(GITHUB_ERROR + " couldn't get reference object");
+    return getCommit(referenceObject.sha());
+  }
+
+  private CompletableFuture<ShaLink> handleSetBlob(final Commit commitResponse,
+      final CommitWrapper commitWrapper, final String content) {
+    commitWrapper.setSha(commitResponse.sha());
+
+    ShaLink tree = commitResponse.tree();
+    com.spotify.github.v3.git.Commit commit = commitResponse.commit();
+    if (tree != null) {
+      commitWrapper.setTreeSha(tree.sha());
+    }
+    if (commit != null) {
+      tree = commit.tree();
+      if (tree == null || tree.sha() == null ) throw new GithubException(GITHUB_ERROR);
+      commitWrapper.setTreeSha(tree.sha());
```

Comment:
this logic seems a bit confusing 🤔 also, would be better to make `ShaLink tree` final and create a new var for the `commit.tree();`

easier to understand if the logic would be smth like
```
if (tree != null) 
       set tree.sha
else 
       set commit.tree.sha() 
```